include/deal.II/differentiation/sd/symengine_tensor_operations.h

```diff
@@ -635,6 +635,121 @@ namespace Differentiation
 
     //@}
 
+    /**
+     * @name Symbol substitution and evaluation
+     */
+    //@{
+
+    /**
+     * Perform a single substitution sweep of a set of symbols into the given
+     * tensor of symbolic expressions.
+     * The symbols in the @p expression_tensor that correspond to the entry keys
+     * of the @p substitution_map are substituted with the map entry's associated
+     * value.
+     * This substitution function may be used to give a set of symbolic
+     * variables either a numeric interpretation or some symbolic definition.
+     *
+     * For more information regarding the performance of symbolic substitution,
+     * and the outcome of evaluation using a substitution map with cyclic
+     * dependencies, see the @ref substitute(const Expression &,
+     * const types::substitution_map &) function.
+     *
+     * @note It is not required that all symbolic expressions be fully resolved
+     * when using this function. In other words, partial substitutions are
+     * valid.
+     */
+    template <int rank, int dim>
+    Tensor<rank, dim, Expression>
+    substitute(const Tensor<rank, dim, Expression> &expression_tensor,
+               const types::substitution_map &      substitution_map);
+
+    /**
+     * Perform a single substitution sweep of a set of symbols into the given
+     * symmetric tensor of symbolic expressions.
```

Comment:
```suggestion
     * SymmetricTensor of symbolic expressions.
```

Reply:
Hmm... I don't think I'm keen to make this change. Not only does this stylistically match the description in the function above for `Tensor`s, but we regularly document functions involving tensors and symmetric tensors in this way. See https://dealii.org/developer/doxygen/deal.II/classSymmetricTensor.html#aa1ba0db3d146ea2d13dca05de9a42c74, for example.